packages/stryker-karma-runner/src/KarmaConfigEditor.ts

```diff
@@ -3,48 +3,43 @@ import * as karma from 'karma';
 import { InputFileDescriptor } from 'stryker-api/core';
 import { ConfigEditor, Config as StrykerConfig } from 'stryker-api/config';
 import KarmaConfigReader from './KarmaConfigReader';
+import { KARMA_CONFIG, KARMA_CONFIG_FILE } from './configKeys';
 
 const log = log4js.getLogger('KarmaConfigEditor');
 
 export default class KarmaConfigEditor implements ConfigEditor {
   edit(strykerConfig: StrykerConfig) {
-    const karmaConfig = new KarmaConfigReader(strykerConfig['karmaConfigFile']).read();
+    // Copy logLevel to local logLevel
+    log4js.setGlobalLogLevel(strykerConfig.logLevel);
+
+    const karmaConfig = new KarmaConfigReader(strykerConfig[KARMA_CONFIG_FILE]).read();
     if (karmaConfig) {
-      KarmaConfigEditor.importFiles(strykerConfig, karmaConfig);
-      KarmaConfigEditor.importDefaultKarmaConfig(strykerConfig, karmaConfig);
+      KarmaConfigEditor.importKarmaConfig(strykerConfig, karmaConfig);
+      KarmaConfigEditor.importFiles(strykerConfig);
     }
   }
 
-  private static importFiles(strykerConfig: StrykerConfig, karmaConfig: karma.ConfigOptions) {
+  private static importFiles(strykerConfig: StrykerConfig) {
+    const karmaConfig: karma.ConfigOptions = strykerConfig[KARMA_CONFIG];
     if (!strykerConfig.files) { strykerConfig.files = []; }
-    if (!karmaConfig.files) { karmaConfig.files = []; }
-    if (!karmaConfig.exclude) { karmaConfig.exclude = []; }
+    if (!Array.isArray(karmaConfig.files)) { karmaConfig.files = []; }
+    if (!Array.isArray(karmaConfig.exclude)) { karmaConfig.exclude = []; }
 
     const files: (karma.FilePattern | string)[] = karmaConfig.files;
```

Comment:
Do we still need this assignment?

Reply:
Yes, we don't no for sure it's an array and we expect arrays for the remainder of this function